test/sample-test/sample_test_launcher.py

```diff
@@ -129,7 +129,9 @@ def _compile(self):
           self._run_pipeline = raw_args['run_pipeline']
 
       if self._run_pipeline:
-        nb_params['experiment_name'] = self._test_name + '-test'
+        if not raw_args['notebook_params']['experiment_name']:
```

Comment:
Could you move all the yaml validation codes in one function, including the yamale validate?